vllm/lora/layers.py

```diff
@@ -309,22 +226,7 @@ def set_lora(
     def forward(self, x: torch.Tensor) -> torch.Tensor:
         added_tokens_mask = x > self.base_layer.org_vocab_size - 1
         embeddings_indices = None
-        if current_platform.is_hpu():
-            embedding_len = self.indices_len[3]
-            # NOTE(vgoel): These asserts can be skipped when upstreaming.
-            # Can be removed from vllm-fork also once lora functionality
-            # on Gaudi stabilizes.
-            if current_platform.is_hpu():
-                emb_len = embedding_len
-                x_shape = x.shape
-                ind_shape = self.embeddings_indices[1].shape
-                assert embedding_len == x.shape[0] * x.shape[1], \
```

Comment:
Can we move these asserts to overloaded function from PunicaWrapper class?

Reply:
this assert cannot be put in the PunicaWrapper class; as self.punica_wrapper.embeddings_indices does not have a access to x